### ISSUE TITLE
jaeger-collector hpa alerting：failed to get memory utilization: missing request for memory

### ISSUE THREAD

**Issue by LeoDemon** [kubesphere/kubesphere#4565]
Versions Used
KubeSphere: 3.2.0
Kubernetes: 1.20.5

Environment
3 master nodes, 2 worker nodes

CentOS Linux release 7.5.1804 (Core)， all node are: 8C/16G

How To Reproduce

1、Go to 'CRD-->ClusterConfiguration-->ks-installer'，modify the "servicemesh.enabled" from "false" to "true"

2、after a few minutes, the pods of istio-system namespace will be generated.

3、check hpa execute status: 
```
[root@master1 ~]# kubectl get hpa -n istio-system -o wide
NAME                   REFERENCE                         TARGETS                        MINPODS   MAXPODS   REPLICAS   AGE
istio-ingressgateway   Deployment/istio-ingressgateway   6%/80%                         1         5         1          14d
istiod-1-11-2          Deployment/istiod-1-11-2          0%/80%                         1         5         1          14d
jaeger-collector       Deployment/jaeger-collector       <unknown>/90%, <unknown>/90%   1         5         1          84m
[root@master1 ~]#
[root@master1 ~]# kubectl describe hpa jaeger-collector -n istio-system
Name:                                                     jaeger-collector
Namespace:                                                istio-system
Labels:                                                   app=jaeger
                                                          app.kubernetes.io/component=hpa-collector
                                                          app.kubernetes.io/instance=jaeger
                                                          app.kubernetes.io/managed-by=jaeger-operator
                                                          app.kubernetes.io/name=jaeger-collector
                                                          app.kubernetes.io/part-of=jaeger
Annotations:                                              <none>
CreationTimestamp:                                        Thu, 23 Dec 2021 12:15:36 +0800
Reference:                                                Deployment/jaeger-collector
Metrics:                                                  ( current / target )
  resource memory on pods  (as a percentage of request):  <unknown> / 90%
  resource cpu on pods  (as a percentage of request):     <unknown> / 90%
Min replicas:                                             1
Max replicas:                                             5
Deployment pods:                                          1 current / 0 desired
Conditions:
  Type           Status  Reason                   Message
  ----           ------  ------                   -------
  AbleToScale    True    SucceededGetScale        the HPA controller was able to get the target's current scale
  ScalingActive  False   FailedGetResourceMetric  the HPA was unable to compute the replica count: failed to get memory utilization: missing request for memory
Events:
  Type     Reason                   Age                    From                       Message
  ----     ------                   ----                   ----                       -------
  Warning  FailedGetResourceMetric  4m11s (x315 over 83m)  horizontal-pod-autoscaler  failed to get memory utilization: missing request for memory
[root@master1 ~]#
```

4、check the jaeger-collector Deployment ，find the container resource does not defined.
```yaml
      containers:
        - name: jaeger-collector
          image: 'jaegertracing/jaeger-collector:1.27'
          ...
          ...
          resources: {}
          volumeMounts:
         ...
         ...
```

please fixed this bug, thanks.

**Comment by FeynmanZhou** [comment]
Hi @RolandMa1986 , could you please help to verify if this is a bug in KubeSphere?

**Comment by RolandMa1986** [comment]
/kind bug
/good-first-issue
Need to add resource'  request and limit to the jaeger cr.
https://github.com/kubesphere/ks-installer/blob/master/roles/ks-istio/templates/jaeger-cr.yaml.j2

**Comment by jioce** [comment]
issue is fixed in v3.3.0-rc.0
